### PR TITLE
mktempfile returns Path object

### PR DESCRIPTION
**Pull request by agusgs**
Related with #12515 

I changed the mktempfile function in IPython/core/interactiveshell.py to return a Path object instead of a string.
Now I'm changing the callers of that function to handle that.

### CODE REVIEW

File: IPython/core/interactiveshell.py [hunk 1]
```diff
@@ -3596,18 +3596,18 @@ def mktempfile(self, data=None, prefix='ipython_edit_'):
...
-        handle, filename = tempfile.mkstemp(".py", prefix, dir=str(dir_path))
-        os.close(handle)  # On Windows, there can only be one open handle on a file
+        temp_directory = Path(tempfile.mkdtemp(prefix=prefix))
+        self.tempdirs.append(temp_directory)
 
+        handle, filename = tempfile.mkstemp('.py', prefix, dir=temp_directory)
         file_path = Path(filename)
-        self.tempfiles.append(file_path)
+        os.close(handle)  # On Windows, there can only be one open handle on a file
+        self.tempfiles.append(file)
```
Comment (Carreau) [inline IPython/core/interactiveshell.py:3605]:
```suggestion
        self.tempfiles.append(file_path)
```

Maybe ?

Comment (agusgs) [reply IPython/core/interactiveshell.py:3605]: Yes, I messed up when rebasing master :)

File: IPython/core/tests/test_interactiveshell.py [hunk 2]
```diff
@@ -561,6 +561,20 @@ def test_exit_code_signal_csh(self):
             else:
                 del os.environ['SHELL']
 
+    def test_removes_temp_files_and_directories(self):
+        # First we ensure that we have created at least a tempfile
+        ip.mktempfile()
+
+        self.system("exit 0")
+        ip.atexit_operations()
+
```
Comment (agusgs) [inline IPython/core/tests/test_interactiveshell.py:570]: @Carreau I'm not sure how to test this case...
I want to be sure that every time a shell is closed we are removing the temp dirs/files.
Any thoughts?

Comment (Carreau) [reply IPython/core/tests/test_interactiveshell.py]: Apologies, I received many PR during hacktoberfest, I'll have a look.

Comment (agusgs) [reply IPython/core/tests/test_interactiveshell.py]: No problem :+1:

**Comment by Carreau** [comment]
Thanks for your patience and sending this; I was not able to keep up with everything and to keep the number of PRs small I'm going to close this one; If you believe this was a mistake, feel free to comment or re-send it; 

I can't guarantee I'll get to it as I do not have the bandwidth.

**Pull request closed without merging**